_posts/2021-04-07-#21061.md

```diff
@@ -7,19 +7,80 @@ authors: [amitiuttarwar]
+discussed rebroadcast in previous review clubs,
+[#16698](https://bitcoincore.reviews/16698) and
+[#18038](https://bitcoincore.reviews/18038).
+
+* The rebroadcast project's goal is to improve privacy by making node
+  rebroadcast behavior for wallet transactions indistinguishable from that of
+other peers' transactions.
+
+* PR#21061 adds a `TxRebroadcast` module responsible for selecting transactions
+  to be rebroadcast and keeping track of how many times each transaction has
+been rebroadcast.  After each block, the module uses the miner and other
+heuristics to select transactions from the mempool that it believes "should"
+have been included in the block and reannounces them (disabled by default for
+now).
+
+* Rebroadcasts happen once per new bock. The set of transactions to be
+  rebroadcast is calculated as follows:
+
+	- The node regularly estimates the minimum feerate for transactions to
+	  be included in the next block, `m_cached_fee_rate`.
+
+	- When a new block arrives, the transactions included in the block are
+	  removed from the mempool. The node then uses `BlockAssembler` to
+calculate which transactions (with a total weight up to 3/4 of the block
+maximum) from the mempool are more than 30 minutes old and have a minimum
+feerate of `m_cached_fee_rate`. The combination of these two filters results in
+a set of transactions that our node would have included in the last block.
+
+	- The rebroadcast attempt tracker, `m_attempt_tracker`, tracks how many
+	  times and how recently we've attempted to rebroadcast a transaction
+so that we don't spam the network with re-announcements.
+
 ## Questions
--->
 
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+What was your review approach?
+
+2. In what scenarios might a user want to rebroadcast their transaction? Why
+shouldn't each wallet just be solely responsible for rebroadcasting its own
+transactions?
+
+3. How does the rebroadcast module decide which transactions to rebroadcast
+(`TxRebroadcastHandler::GetRebroadcastTransactions()`)?
+
+4. In what scenarios would a miner include different transactions from our
+`BlockAssembler`? More specifically, when might the miner exclude a
+transaction, and when might it include a transaction yours doesn't?
+
+5. Why might we want to keep a transaction in our rebroadcast attempt tracker
+even after removing it from our mempool?  (Hint: what happens if we expire a
+transaction from our mempool and then our peer rebroadcasts it to us? When
+might this happen?)
+
+6. When should we remove transactions from our rebroadcast attempt tracker? How
+does the code ensure that the tracker doesn't grow unbounded?
+
+7. How is the estimated minimum feerate for inclusion in a block,
+`m_cached_fee_rate`, calculated? Why not just calculate the feerate of the
+lowest-feerate transaction in the block?
```

Comment:
```suggestion
lowest-feerate transaction in the received block?
```

chain tip? network block? I'm not sure of the wording but a clarification could be useful.